Lib/importlib/_bootstrap_external.py

```diff
@@ -910,6 +910,7 @@ def __init__(self, fullname=None, path=None, *, name=None):
         if name is not None:
             _warnings.warn("the 'name' parameter is deprecated; use "
                            "'fullname' instead", DeprecationWarning)
+            self.name = name
         self.name = fullname
```

Comment:
You need to put `self.name = fullname` in an `else:` statement, otherwise you'll *always* re-assign `self.name` to `fullname` anyway.